docsource/use_cases/module_renaming.rst

```diff
@@ -0,0 +1,169 @@
+if you are working on a version of OpenUpgrade under development.
+
+Coverage File
+"""""""""""""
+
+A coverage file is present in each version:
+
+* Until version 13, the file is present in this path:
+  ``odoo/openupgrade/doc/source/modulesXXX-YYY.py``.
+
+* from version 14.0 onwards, the file is present in this path:
+  ``docsource/modulesXXX-YYY.py``.
+
+Edit this file, and add the following text:
+
+.. code-block:: rst
+
+    +-----------------------------------+-------------------------------------------------+
+    | |new| crm_iap_enrich              | Renamed from crm_iap_lead_enrich                |
+    +-----------------------------------+-------------------------------------------------+
+    | |del| crm_iap_lead_enrich         | Renamed to crm_iap_enrich                       |
+    +-----------------------------------+-------------------------------------------------+
+
+Generate / Update analysis file
+"""""""""""""""""""""""""""""""
+
+As long as the renaming has not been identified, the ``upgrade_analysis.txt`` generated
+at the initialization of the OpenUpgrade version is partially wrong, because openupgrade
+considers that the xml_id of the module are new, while they can come from the module of
+the previous version.
+
+For the modules that concern Odoo, you must therefore update the version analysis
+with the upgrade_analysis tool present in the OCA/server-tools project.
+
+Once done, the analysis file contains correct information, and can be studied to
+write remaining pre / post migration scripts.
+
+
+.. code-block:: xml
+
+    ---XML records in module 'crm_iap_enrich'---
+    NEW ir.actions.server: crm_iap_enrich.action_enrich_mail [renamed from crm_iap_lead_enrich module]
+    DEL ir.actions.server: crm_iap_lead_enrich.action_enrich_mail [renamed to crm_iap_enrich module]
+
+See `Full v15 upgrade analysis file <https://github.com/OCA/OpenUpgrade/blob/97491cb7d9a8ed494a49cf1db9b7fc8852aac254/openupgrade_scripts/scripts/crm_iap_enrich/15.0.1.1/upgrade_analysis.txt>`_.
+
+Notes
+-----
+
+* In the ``apriori.py`` file, in the ``renamed_modules`` dict, put the Odoo renaming at
+  the beginning, ordered by module name, then OCA modules renaming, with the mention of
+  OCA repository. Example:
+
+.. code-block:: python
+
+    renamed_modules = {
+        # Odoo
+        'crm_reveal': 'crm_iap_lead',
+        'payment_ogone': 'payment_ingenico',
+        # OCA/delivery-carrier
+        'delivery_carrier_label_ups': 'delivery_ups_oca',
+        # OCA/event
+        'website_event_filter_selector': 'website_event_filter_city',
+    }
+
+* If your instance has custom modules, and if you took advantage of the migration
+  to refactor these custom modules by renaming them,
+  you can add the following lines at the end of this file:
+
+.. code-block:: python
+
+    renamed_modules.update({
+        "my_obsolete_module": "my_renamed_module",
+    })
+
+(no need in this case to propose a PR to the community)
```

Comment:
Very good idea. I didn't use this trick and I constantly run with merge conflict when upgrading the `apriory.py` file. Thanks for the advice. :)